lightcycle-processor/src/test/java/com/soundcloud/lightcycle/ActivityInjectionTest.java

```diff
@@ -0,0 +1,259 @@
+package com.soundcloud.lightcycle;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+
+import org.junit.Test;
+
+import javax.tools.JavaFileObject;
+
+import static com.google.common.truth.Truth.assertAbout;
+import static com.google.testing.compile.JavaFileObjects.forSourceString;
+import static com.google.testing.compile.JavaSourceSubjectFactory.javaSource;
+import static com.google.testing.compile.JavaSourcesSubjectFactory.javaSources;
+
+/**
+ * Created by yuwono-niko on 11/9/16.
+ */
+
+public class ActivityInjectionTest {
+
+    private static final String VALID_TEST_ACTIVITY = Joiner.on("\n").join(
+            "package com.test;",
+            "",
+            "import com.soundcloud.lightcycle.ActivityLightCycle;",
+            "import com.soundcloud.lightcycle.DefaultActivityLightCycle;",
+            "import com.soundcloud.lightcycle.LightCycle;",
+            "import com.soundcloud.lightcycle.LightCycleDispatcher;",
+            "",
+            "import android.app.Activity;",
+            "",
+            "public class ValidTestActivity extends Activity implements LightCycleDispatcher<ActivityLightCycle<ValidTestActivity>> {",
+            "",
+            "    @LightCycle LightCycle1 lightCycle1;",
+            "    @LightCycle LightCycle2 lightCycle2;",
+            "",
+            "    @Override",
+            "    public void bind(ActivityLightCycle<ValidTestActivity> lightCycle) {",
+            "        // nop",
+            "    }",
+            "",
+            "}",
+            "",
+            "class LightCycle1 extends DefaultActivityLightCycle<ValidTestActivity> {",
+            "}",
+            "",
+            "class LightCycle2 extends DefaultActivityLightCycle<ValidTestActivity> {",
+            "}");
+
+    private static final String ACTIVITY_BINDER_SRC = Joiner.on("\n").join(
+            "package com.test;",
+            "",
+            "public final class ValidTestActivity$LightCycleBinder {",
+            "",
+            "    public static void bind(ValidTestActivity target) {",
+            "        final com.soundcloud.lightcycle.ActivityLightCycle<com.test.ValidTestActivity> lightCycle1$Lifted = com.soundcloud.lightcycle.LightCycles.lift(target.lightCycle1);",
+            "        target.bind(lightCycle1$Lifted);",
+            "        final com.soundcloud.lightcycle.ActivityLightCycle<com.test.ValidTestActivity> lightCycle2$Lifted = com.soundcloud.lightcycle.LightCycles.lift(target.lightCycle2);",
+            "        target.bind(lightCycle2$Lifted);",
+            "    }",
+            "}");
+
+    @Test
+    public void shouldGenerateInjectorForActivity() {
+        JavaFileObject validTestActivity = forSourceString("com/test/ValidTestActivity", VALID_TEST_ACTIVITY);
```

Comment:
I was thinking about inlining them too, but since the original code use constants I thought it'll be better to follow the existing code style. 

But yes I agree with you, it'll look better if these constants are inlined. Will change it.